### PR TITLE
Fix smart varialbes tests.

### LINKED ISSUE #3938: Fix smart variables test failures
IMO current picture of failures for smart variables is a bit too much and worth analysis:

| Module | Tier 1 | Tier 2 | Total |
| --- | --- | --- | --- |
| API | 30 | 3 | 33 |
| CLI | 4 | 5 | 9 |
| UI | 11 | 18 | 29 |
|  |  |  | _71_ |

### PR DESCRIPTION
**Pull request by svtkachenko**
This change should fix 44 failures for `ui/test_variables` and 15 failures for `cli/test_variables`. Closes #3938 and also #3931.

```python
 py.test -n 2 -v tests/foreman/ui/test_variables.py
=========================================================== 1 failed, 43 passed in 2848.72 seconds ============================================================
```
1 failure looks more like a problem with my local infrastructure, after re-run it passed.
```python
py.test -v tests/foreman/ui/test_variables.py -k 'test_positive_update_name'
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.31, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
tests/foreman/ui/test_variables.py::SmartVariablesTestCase::test_positive_update_name <- robottelo/decorators/__init__.py PASSED
===================================================================== 43 tests deselected =====================================================================
========================================================== 1 passed, 43 deselected in 237.46 seconds ==========================================================
```

### CODE REVIEW

File: robottelo/ui/hosts.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def get_yaml_output(self, name):
         self.wait_for_ajax()
         return output
 
-    def get_smart_variable_value(self, host_name, sv_name, hidden=False):
+    def get_smart_variable_value(self, host_name, sv_name):
```
Comment (oshtaier) [inline robottelo/ui/hosts.py]: fix docstring as well

File: robottelo/ui/hosts.py [hunk 2]
```diff
@@ -220,14 +220,11 @@ def get_smart_variable_value(self, host_name, sv_name, hidden=False):
...
-            locator = locators['host.smart_variable_value_hidden'] % sv_name
-        else:
-            locator = locators['host.smart_variable_value'] % sv_name
-        return self.wait_until_element(locator)
+        return self.wait_until_element(
+            locators['host.smart_variable_value'] % sv_name)
 
     def set_smart_variable_value(self, host_name, sv_name, sv_value,
-                                 override=True, hidden=False):
+                                 override=True):
```
Comment (oshtaier) [inline robottelo/ui/hosts.py]: fix docstring as well

**Comment by oshtaier** [comment]
Provide test results for CLI as well

**Comment by oshtaier** [comment]
basically, PR is really good

**Comment by oshtaier** [comment]
Not sure about leaving commented code, even when it makes sense

**Comment by oshtaier** [comment]
Pending CLI test results

**Comment by svtkachenko** [comment]
Docstrings updated.  Plus a few missing changes for CLI.
Run results for CLI.
```python
% nosetests tests/foreman/cli/test_variables.py
..SSSS....S..S......SSSSS.....S..................
----------------------------------------------------------------------
Ran 49 tests in 1119.700s
OK (SKIP=12)
```

**Pull request merged**